torch/lib/c10d/TCPStore.hpp

```diff
@@ -124,12 +125,25 @@ class TCPStoreWorkerDaemon : public BackgroundThread {
   bool callbackRegisteredData_ = false;
 };
 
+struct TCPStoreOptions {
```

Comment:
Curious: does it make sense to have a constructor versus the bracket-based initialization that's used? Not sure about the advantages/disadvantages there.

Reply:
Since all fields have default values (meaning they have "optional" semantic) you do not gain anything by introducing a constructor. Aggregate initialization such as:

```
TCPStoreOptions opts{/* port */ 1234 };
opts.isServer = true;
```

is syntactically identical to what you would get with a constructor.

Beyond that the whole reason of an "options" struct is to avoid having a constructor with a long list of parameters.